### ISSUE TITLE
Problems with invoke in idispatch_windows.go with *string parameters

### ISSUE THREAD

**Issue by dsmcfarl** [go-ole/go-ole#115]
I ran into an issue with passing a *string parameter to CallMethod of IDispatch. The parameter which is used to return a string value from the COM method call was not being updated as it should.  I traced it down to the invoke implementation in idispatch_windows.go.  I saw that the code that updates output *string parameters was commented out. Uncommenting the code returned gibberish, however, two simple modifications made it return the correct strings:

Original:
```
    for _, varg := range vargs {                                                
        if varg.VT == VT_BSTR && varg.Val != 0 {                                
            SysFreeString(((*int16)(unsafe.Pointer(uintptr(varg.Val)))))        
        }                                                                       
        /*                                                                      
            if varg.VT == (VT_BSTR|VT_BYREF) && varg.Val != 0 {                 
                *(params[n].(*string)) = LpOleStrToString((*uint16)(unsafe.Pointer(uintptr(varg.Val))))
                println(*(params[n].(*string)))                                 
                fmt.Fprintln(os.Stderr, *(params[n].(*string)))                 
            }                                                                   
        */                                                                      
    }  
```

New:
```
    for i, varg := range vargs { 
       n := len(params) - i - 1                                        
        if varg.VT == VT_BSTR && varg.Val != 0 {                                
            SysFreeString(((*int16)(unsafe.Pointer(uintptr(varg.Val)))))        
        }                                                               
        if varg.VT == (VT_BSTR|VT_BYREF) && varg.Val != 0 {                 
            *(params[n].(*string)) = LpOleStrToString(*(**uint16)(unsafe.Pointer(uintptr(varg.Val))))      
        }                                                                
    }  
```

However, I suspect that that is just a patch and the real issue is that LpOleStrToString should be updated to accept a **uint16 (a pointer to an array of wide characters) instead of a *uint16 (an array of wide characters) and if needed the current implementation of LpOleStrToString should be renamed OleStrToString.

I've only been working with the code for a day so I may be way off. If this approach sounds right, I'd be happy to create a pull request.

**Comment by jacobsantos** [comment]
Please do create a pull request. If you may, could you include test cases? I would write them if you don't, but it would be extremely helpful if you provided them.

**Comment by jacobsantos** [comment]
Fixed in #116.
